### ISSUE TITLE
Open Community Working Meeting 2022-06-03

### ISSUE THREAD

**Issue by Relequestual** [json-schema-org/community#183]
📺 [See Recording](https://postman.zoom.us/rec/share/jTcXUOgjF0N5CFouqEK5uIKXHxCzN8zySQhFu9ab_n-Tc6FuBTHsrZ_AsRCJg2J3.pWzEGCNCf3sM2Yy9?startTime=1654282518000)

⏪ [Go To Previous Meeting](https://github.com/json-schema-org/community/issues/177)

## Agenda

```
No agenda items were set for this meeting
```

## Highlights

- @jviotti shared his project for upgrading and downgrading JSON Schemas between versions/drafts. [AlterSchema](https://github.com/jviotti/alterschema) which is designed to be langauge agnostic.

- There was a good discussion around schema version compatibility from @jdesrosiers and @awwright.

- Further discussion on vocabularies and how they could look in the future.

- A discussion on keywords that depend on other keywords

- A discussion on finite state machines, push down automata and ABNF's applicability in general and to JSON Schema. 
 
## Actions

- [ ]  @Relequestual to have a backup for scheduled calls when on a holiday. 

- [ ]  @jdesrosiers to write a blog post about concerning additionalProperties, inheritence concept etc.

## Attendees

| Account           |
|:-----------------:|
| @awwright         |
| @jdesrosiers      |
| @jviotti          |
| Vipul             |
| Pranit Brahmbhatt |
| @gregsdennis      |
| @handrews         |

### Upgrading and downgrading `Schema` versions using `Alter Schema`

Alterschema is trying to solve upgrading and downgrading between JSON Schema and both OpenAPI versions. i.e how given any json schema can you upgrade it and can you also downgraded in a lossy manner. 

As it was noticed ( while working at Postman ) that people are stuck with `draft 4` and `draft 6`. We realized that people do want to upgrade but the tooling's support for older versions becoming a barrier.

---

### Fragmented ecosystem and compatibility.

Working group discussed how the ecosystem is fragmented. An example presented had to do with differences in semantic understanding while in usage the behaviour is expected with original meaning i.e of the JSON Schema specification. This leads to differences in implementation themselves i.e behaviour changes. Another is certain limitations eg. limited amount of memory leading changes in implementation and compatibility.

There is an understanding that the vocabulary system is to address those problems in the future. Also, it is understood that idea of vocabularies should be malleable where it's a collection of keywords that are defined so that defining keywords and collecting them into a group, can be called a vocabulary.

Regarding keywords that depend on other keywords, there were two points raised of resuability and usage of multiple keywords at a time and the annotation for behavior purposed of the same. The current proposal of the working group is already understood to moving in the direction of ironing it out.

---

Further Readings

- https://github.com/awwright/jsonschemaparse
- https://github.com/jviotti/alterschema
- https://json-e.js.org
- https://www.inkandswitch.com/cambria/
- https://github.com/awwright/jsonschemaparse
- https://www.jsonbinpack.org
- https://github.com/json-schema-org/json-schema-spec/wiki/Profiles-and-Sets
- https://sourcemeta.github.io/json-size-benchmark/
- https://arxiv.org/abs/2201.02089

**Comment by benjagm** [comment]
Closing this issue as all tasks are completed. Thanks for your contributions!

**Comment by benjagm** [comment]
@jdesrosiers are you still considering to write a blog post about concerning additionalProperties and inheritence concept?

**Comment by jdesrosiers** [comment]
No, not in the near future. I don't even remember what this was in reference to and I don't see myself having time for this any time soon.
